Library/Homebrew/cask/quarantine.rb

```diff
@@ -18,7 +18,17 @@ module Quarantine
     COPY_XATTRS_SCRIPT = (HOMEBREW_LIBRARY_PATH/"cask/utils/copy-xattrs.swift").freeze
 
     def self.swift
-      @swift ||= DevelopmentTools.locate("swift")
+      @swift ||= begin
+        # /usr/bin/swift (which runs via xcrun) adds `/usr/local/include` to the top of the include path,
+        # which allows really broken local setups to break our Swift usage here. Using the underlying
+        # Swift executable directly however (returned by `xcrun -find`) avoids this CPATH mess.
+        xcrun_swift = ::Utils.popen_read("/usr/bin/xcrun", "-find", "swift", err: :close).chomp
+        if $CHILD_STATUS.success? && File.executable?(xcrun_swift)
+          xcrun_swift
+        else
+          DevelopmentTools.locate("swift")
+        end
+      end
     end
```

Comment:
It looks the old method returns a Pathname here but all the usages seem to just get coerced to strings so it's probably fine. On that note, it might be good to add types to this method.

Before:

```rb
brew(main):001:0> Cask::Quarantine.send(:swift)
=> #<Pathname:/usr/bin/swift>
```

After:

```rb
brew(main):001:0> Cask::Quarantine.send(:swift)
=> "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift"
```

Reply:
Yeah maybe will keep it as a string because it's very slightly cheaper